BattleNetwork/overworld/bnOverworldOnlineArea.cpp

```diff
@@ -1219,32 +1234,35 @@ void Overworld::OnlineArea::sendEmoteSignal(const Overworld::Emotes emote)
   packetProcessor->SendPacket(Reliability::Reliable, buffer);
 }
 
-void Overworld::OnlineArea::sendObjectInteractionSignal(unsigned int tileObjectId)
+void Overworld::OnlineArea::sendObjectInteractionSignal(unsigned int tileObjectId, uint8_t button)
```

Comment:
Why not use an InputEvent struct instead of button?

Reply:
This is the signal being sent directly to the server (voted api only cares about ints), InputEvents are strings and might be more likely to change from something like renaming the event.

Reply:
Another reason is I can resolve the int once higher up, rather than at every signal

Reply:
what about something like:

```cpp
sendObjectInteractionSignal(tileObjectId, InputEvent::FromServerButton(button));
```

Reply:
actually it might be wise to have some sort of `ServerBridge::From()` utils like this instead of baking it into the other classes. 

`sendObjectInteractionSignal(tileObjectId, ServerBridge::InputEventFrom(button));`

Reply:
Handling it this way would mean including guards such as: 

```cpp
void Overworld::Homepage::OnInteract(const InputEvent& event) {
  if(event != InputEvents::pressed_interact) {
    return;
  }
  // ... etc
```

I'm switching the code to this currently if you're ok with this.

Reply:
The last part that bothers me is: if we decide to switch what input events we're using such as using a new talk_to_navi input event instead of shoulder_left, code can compile despite completely breaking (testing for shoulder_left but it is never sent, but talk_to_navi and interact are)

Reply:
Good point. I'd want some way to describe what "button" is supposed to be but without tying it to an actual button that we have to come up with. What if we just send an `enum` over the network? Something that doesn't restrict users with "buttons" but their intentions. e.g.

```cpp
enum class InteractType : char {
  inspect = 0,
  player, // initiate an interaction with the player (e.g. quest hints)
  action // talk, press button, "actionable" things
}
```